virttest/qemu_vm.py

```diff
@@ -104,14 +104,20 @@ def qemu_proc_term_handler(vm, monitor_exit_status, exit_status):
     Callback function to detect QEMU process non-zero exit status and
     push VMExitStatusError to background error bus.
 
-    :param vm: VM name.
+    :param vm: VM object.
     :param monitor_exit_status: True to push VMUnexpectedExitError instance
         with calltrace to global error event bus.
     :param exit_status: QEMU process exit status.
     """
+    for _ in vm.devices.temporary_image_snapshots[::-1]:
+        try:
+            os.unlink(vm.devices.temporary_image_snapshots.pop())
+        except OSError:
```

Comment:
Since `os.unlink` would raise different types of exceptions between py2 and py3, suggest catching all kinds of errors instead.
```
Python 3.7.6 (default, Jan 30 2020, 09:44:41) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.unlink('/tmp/xxx')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/xxx'
```
```
Python 2.7.17 (default, Oct 20 2019, 00:00:00) 
[GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.unlink('/tmp/xxx')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 2] No such file or directory: '/tmp/xxx'
```

Reply:
> Since `os.unlink` would raise different types of exceptions between py2 and py3, suggest catching all kinds of errors instead.
> 
> ```
> Python 3.7.6 (default, Jan 30 2020, 09:44:41) 
> [GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux
> Type "help", "copyright", "credits" or "license" for more information.
> >>> import os
> >>> os.unlink('/tmp/xxx')
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
> FileNotFoundError: [Errno 2] No such file or directory: '/tmp/xxx'
> ```
> 
The `FileNotFoundError` is inherited from `OSError`.
```

      +-- OSError
      |    +-- BlockingIOError
      |    +-- ChildProcessError
      |    +-- ConnectionError
      |    |    +-- BrokenPipeError
      |    |    +-- ConnectionAbortedError
      |    |    +-- ConnectionRefusedError
      |    |    +-- ConnectionResetError
      |    +-- FileExistsError
      |    +-- FileNotFoundError
```
I think will catch it.

```
Python 3.7.4 (default, Aug 12 2019, 14:45:07) 
[GCC 9.1.1 20190605 (Red Hat 9.1.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> try:
...     os.unlink('/tmp/xxxxxx')
... except OSError:
...     pass
... 
>>> 
>>> 
```



> ```
> Python 2.7.17 (default, Oct 20 2019, 00:00:00) 
> [GCC 9.2.1 20190827 (Red Hat 9.2.1-1)] on linux2
> Type "help", "copyright", "credits" or "license" for more information.
> >>> import os
> >>> os.unlink('/tmp/xxx')
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
> OSError: [Errno 2] No such file or directory: '/tmp/xxx'
> ```

Reply:
@YongxueHong uh, yes, thanks for your explanation!